bcipy/signal_model/mach_learning/train_model.py

```diff
@@ -55,6 +54,70 @@ def train_pca_rda_kde_model(x, y, k_folds=10):
     model.fit(x, y)
 
     # Report AUC
-    print('AUC-i: {}, AUC-cv: {}'.format(auc_init, auc_cv))
+    print('AUC-i: {}, AUC-cv: {}'.format(auc_init, model.last_cv_auc))
+
+    return model
+
+
+def train_m_estimator_pipeline(x, y):
+    """ Trains Cw-m-PCA m-RDA KDE model given the input data and labels, returns the model
+        Args:
+            x(ndarray[float]): C x N x p data array
+            y(ndarray[int]): N x 1 observation (class) array
+                N is number of samples p is dimensionality of features
+                C is number of channels
+            k_folds(int): Number of cross validation folds
+            cross_validate(Boolean): True if cross validation AUC is to be calculated. Increases required time.
+        Return:
+            model(pipeline): trained model
+    """
+
+    # Train PCA and MDA to cross validate hyper parameters
+    model = Pipeline()
+
+    pca = MPCA(var_tol=.1**5)
+    mda = MDiscriminantAnalysis()
+
+    model.add(pca)
+    model.add(mda)
+
+    lam, gam = cross_validate_parameters(x=x, y=y, model=model)
+    cv_auc = model.last_cv_auc
+    print('Optimized val [lam:{} \ gam:{}]'.format(lam, gam))
+
+    # Train the model with optimized hyper parameters to find once again the CV AUC
+    model = Pipeline()
+
+    pca = MPCA(var_tol=.1**5)
+    mda = MDiscriminantAnalysis()
+
+    model.add(pca)
+    model.add(mda)
+
+    model.pipeline[1].lam = lam
+    model.pipeline[1].gam = gam
+
+    sc = model.fit_transform(x, y)
+    fpr, tpr, _ = metrics.roc_curve(y, sc, pos_label=1)
+    auc_all = metrics.auc(fpr, tpr)
+
+    bandwidth = 1.06 * min(np.std(sc), iqr(sc) / 1.34) * np.power(x.shape[0], -0.2)
+
+    model = Pipeline()
+
+    pca = MPCA(var_tol=.1**5)
```

Comment:
Same, these should be top level variables

Reply:
I'm not even sure what do you mean by 'Same, these should be top level variables' , same with what? top level of what? This pca object is different than regular methods pca object, they cant be shared.